### ISSUE TITLE
Add pulse count measure for tasmota in some GPIO with some configured divider

### ISSUE THREAD

**Issue by roblad** [arendst/Sonoff-Tasmota#1188]
Hi,

Is it possible to add some pulse count measure as an imput GPIO pin for i.e. water or energy pulse measure

i.e.

https://github.com/rogerclarkmelbourne/arduino-esp8266/blob/master/esp8266/cores/esp8266/core_esp8266_wiring_pulse.c

http://www.instructables.com/id/Wearable-heart-beat-sensor-ESP8266Pulse-sensor/

https://learn.openenergymonitor.org/electricity-monitoring/pulse-counting/interrupt-based-pulse-counter

https://community.openenergymonitor.org/t/wemos-d1-mini-esp8266-pulse-counting/4405

**Comment by davidelang** [comment]
I believe that something like that was already added.

David Lang

**Comment by roblad** [comment]
Hi,

Do U know what is a description in soft for that ?

**Comment by ionciubotaru** [comment]
Red about counters https://github.com/arendst/Sonoff-Tasmota/wiki/Commands#sensor

**Comment by roblad** [comment]
Hi,


I see


CounterType<x> |   | Show current Counter 1..4 type as pulse Counter or pulse Timer
-- | -- | --
CounterType<x> | 0 / off | Set current Counter 1..4 type as pulse Counter
CounterType<x> | 1 / on | Set current Counter 1..4 type as pulse Timer



but haw to usage, I need to set one GPIO as counter 1...4 and count the value ? I need some divider i.e. for each 50 counts 1 send counted value as 1

**Comment by davidelang** [comment]
the sonoff will report the count, you do the math in your home automation 
software

**Comment by myozone** [comment]
Yes two MQTT outputs would be very useful one for an incremented count and the other time between pulses for a energy or water meter instantaneous watts and Kwh

**Comment by roblad** [comment]
see that

https://github.com/stefanbode/Sonoff-Tasmota/issues/15
